codedoc.c

```diff
@@ -4239,6 +4241,8 @@ scan_file(filebuf_t   *file,		/* I  - File to scan */
   }
 
   mxmlDelete(comment);
+  mxmlDelete(function);
```

Comment:
I'm also not sure that this is the right thing to do - if the function is getting documented then we don't want to delete the node.

Also, leaking one node isn't a big deal given the limited lifespan of an instance of codedoc and its intended uses.